clients/da-vinci-client/src/main/java/com/linkedin/davinci/schema/merge/SortBasedCollectionFieldOpHandler.java

```diff
@@ -128,10 +128,10 @@ public UpdateResultStatus handlePutList(
     newElements.addAll(toPutList);
 
     // Add elements and timestamps for the collection-merge part.
-    activeElementToTsMap.forEach((activeElement, activeTimestamp) -> {
-      newElements.add(activeElement);
-      newActiveTimestamps.add(activeTimestamp);
-    });
+    for (Map.Entry<Object, Long> entry: activeElementToTsMap.entrySet()) {
+      newElements.add(entry.getKey());
+      newActiveTimestamps.addPrimitive(entry.getValue().longValue());
```

Comment:
Might be a dumb question but just wanted to confirm: So the entry.getValue() is already Long, that means it was already boxed to Long before. Hmm then why changing to this addPrimitive(long) API could save here?

>  public boolean add(Long o) {
    return addPrimitive(o);
  }

Reply:
Yeah in that case it shouldn't change anything. There was one specific code path that seemed to account for a large fraction of allocations (25% IIRC!) due to Long boxing. I also changed the other code paths that use the `PrimitiveLongList` for the sake of consistency but the other paths may not have as much impact.

Reply:
I see. Thanks for the change!